pkg/virt-launcher/virtwrap/network/podinterface.go

```diff
@@ -298,12 +298,20 @@ func retrieveMacAddress(iface *v1.Interface) (*net.HardwareAddr, error) {
 	return nil, nil
 }
 
-// The only difference between bindings for two phases is that the first phase
-// should not require access to domain definition, hence we pass nil instead of
-// it. This means that any functions called under phase1 code path should not
-// use the domain set on the binding.
-func getPhase1Binding(vmi *v1.VirtualMachineInstance, iface *v1.Interface, network *v1.Network, podInterfaceName string) (BindMechanism, error) {
-	return getPhase2Binding(vmi, iface, network, nil, podInterfaceName)
+func getPhase1Binding(vmi *v1.VirtualMachineInstance, iface *v1.Interface, network *v1.Network, podInterfaceName string, launcherPID int) (BindMechanism, error) {
+	if iface.Bridge != nil {
+		return generateBridgeVMNetworkingConfigurator(vmi, iface, podInterfaceName, launcherPID)
+	}
+	if iface.Masquerade != nil {
+		return generateMasqueradeVMNetworkingConfigurator(vmi, iface, network, podInterfaceName, launcherPID)
+	}
+	if iface.Slirp != nil {
+		return &SlirpBindMechanism{vmi: vmi, iface: iface, launcherPID: launcherPID}, nil
```

Comment:
No. Despite the current design, slirp is **not** a bind mechanism - it implements a single method of the bind mechanism interface. 

As such, I'm pointing my finger at it from the beginning. Once I finish refactoring this package, this will implement an interface named (something like) `LibvirtInterfaceSpecGenerator` instead, and implement the single method container in it: `generate() error` .